File: Arrays/414-third-maximum-number/414-third-maximum-number.cpp
class Solution {
public:
    int thirdMax(vector<int>& nums) {
        set<int> s(nums.begin(),nums.end());
        vector<int> res{s.begin(),s.end()};
        return res.size()<3 ? *res.rbegin() : *(res.rbegin() + 2);
    }
};

File: Arrays/414-third-maximum-number/414-third-maximum-number.py
class Solution:
    def thirdMax(self, nums: List[int]) -> int:
        num=set(nums)
        num=list(num)
        num.sort(reverse=True) 
        if (len(num)<3):
            return num[0]
        else:
            return num[2]
       
        